### DISCUSSION TITLE
Support for select optgroup tag

### DISCUSSION THREAD

**Discussion by 0hra** [livewire/flux#281] [Feature Requests]
Add support for optgroup in Select component?
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/optgroup

I can wrap flux options in `<optgroup label="name"></optgroup>` tags, but I don't know... maybe add a flux-tag for it as well?

**Comment by cladico** [comment]
It works to some extent, but when searching, all the “optgroup” headers/options show up, here is a duplicate conversation #354 .
I tried this approach:

```blade
<flux:select
    variant="listbox"
    searchable
    placeholder="City"
    wire:model.live="cityId"
    label="Location"
>
    <x-slot name="search">
        <flux:select.search wire:model.live.debounce.500ms="searchTerm" class="px-4" placeholder="Search city..." />
    </x-slot>

    @foreach ($groupedCities as $countyId => $cities)
        @if (!empty($cities) && isset($counties[$countyId]))
            <flux:select.option disabled>
                <div class="flex items-center gap-2">
                    <flux:icon.arrow-down-right variant="mini" class="text-zinc-400" />
                    {{ $counties[$countyId] }}
                </div>
            </flux:select.option>

            @foreach ($cities as $city)
                <flux:select.option :selected="$city['id'] == $cityId" value="{{ $city['id'] }}">
                    {{ $city['name'] }}
                </flux:select.option>
            @endforeach
        @endif
    @endforeach
</flux:select>
```

I’m using `<flux:select.option disabled>` as a workaround for `<optgroup label="...">`, but it isn’t reliable, sometimes it behaves incorrectly. Proper `<optgroup>` support exists for a practical reason (e.g., cities grouped by county), and this workaround doesn’t fully replace it.
I upgraded to FluxUI Pro to avoid spending time on this kind of issue, but I ended up losing several hours here.
